qiskit/transpiler/passes/routing/layout_transformation.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2018.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Map (with minimum effort) a DAGCircuit onto a `coupling_map` adding swap gates."""
+
+from qiskit.transpiler.basepasses import TransformationPass
+from qiskit.transpiler.exceptions import TranspilerError
+from qiskit.transpiler.routing import util
+from qiskit.transpiler.routing.general import ApproximateTokenSwapper
+
+
+class LayoutTransformation(TransformationPass):
+    """ Adds a Swap circuit for a given (partial) permutation to the circuit.
+
+    This circuit is found by a 4-approximation algorithm for Token Swapping.
+    More details are available in the routing code.
+    """
+
+    def __init__(self, coupling_map, initial_layout, final_layout, trials=4):
+        """LayoutTransformation initializer.
+
+        Args:
+            coupling_map (CouplingMap): Directed graph represented a coupling map.
+            initial_layout (Layout): The starting layout of qubits onto physical qubits.
+            final_layout (Layout): The final layout of qubits on phyiscal qubits.
+            trials (int): How many randomized trials to perform, taking the best circuit as output.
+        """
+        super().__init__()
+        self.coupling_map = coupling_map
```

Comment:
Handling the case `coupling_map=None`. I prefer `raise TranspilerError` (to generate a complete graph) for now.

Reply:
Do you mean I should raise a TranspilerError if no coupling_map is given?

Reply:
Yes

Reply:
Oh, I see what you mean. It is not necessary to do so, because we already declare that None is not allow for coupling_map by not providing default value.

Reply:
Oh, I see what you mean and you are right. It is not necessary to do so, because we already declare that None is not allow for coupling_map by not providing default value. And other passes which take coupling_map usually don't check None, so we should follow the convention.